internal/db/load.go

```diff
@@ -0,0 +1,140 @@
+package db
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/json-iterator/go"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+)
+
+type object struct {
+	Data map[string]interface{} `json:"data"`
+	Type interface{}            `json:"type"`
+}
+
+func loadFromSrc(tx RWTx) error {
+	var files []string
+	filepath.Walk("src/",
+		func(path string, info os.FileInfo, err error) error {
+			if info.IsDir() {
+				return nil
+			}
+			files = append(files, path)
+			return nil
+		})
+	for _, file := range files {
+		data, err := ioutil.ReadFile(file)
+		if err != nil {
+			return err
+		}
+		err = loadFromJSON(tx, data)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func loadFromJSON(tx RWTx, rawO []byte) error {
+	var o object
+	jsoniter.Unmarshal(rawO, &o)
+	obj, e := parseObject(o.Type, o.Data)
+	if e != nil {
+		return e
+	}
+	switch obj.(type) {
+		case Code:
+			c := obj.(Code)
+			r := RecordForModel(CodeModel)
+			SaveCode(r, c)
+			tx.Insert(r)
+		case Datatype:
+			d := obj.(Datatype)
+			r := RecordForModel(CodeModel)
+			SaveCode(r, d.Validator)
+			tx.Insert(r)
+			r = RecordForModel(DatatypeModel)
+			SaveDatatype(r, d)
+			tx.Insert(r)
+	}
+	return nil
+}
+
+func parseObject(t interface{}, data map[string]interface{}) (interface{}, error) {
+	if _, ok := t.(string); !ok {
+		return nil, fmt.Errorf("Type field not found in JSON object")
+	}
+	switch t.(string) {
+	case "code":
+		return parseCode(data)
+	case "datatype":
+		return parseDatatype(data)
+	default:
+		return nil, fmt.Errorf("Unrecognized type %s found", t)
+	}
+}
+
+func parseDatatype(v interface{}) (Datatype, error) {
```

Comment:
all of this manual parsing from json could probably just be replaced by struct tags.. let's talk tomorrow about how to do that!